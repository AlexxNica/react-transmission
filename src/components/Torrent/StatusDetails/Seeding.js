import React from 'react';
import { FormattedMessage } from 'react-intl';

import { formatUL } from 'util/formatters';

export default ({ torrent: { peersGettingFromUs, peersConnected, rateUpload } }) => (
  <FormattedMessage
    id='torrent.full.seeding'
    defaultMessage={`Seeding to {peersGettingFromUs, number} of {peersConnected, number} connected {peersConnected, plural, one{peer} other{peers}} - {formattedRateUpload}`}
    values={{
      peersGettingFromUs,
      peersConnected,
      formattedRateUpload: formatUL(rateUpload),
    }}
  />
);
