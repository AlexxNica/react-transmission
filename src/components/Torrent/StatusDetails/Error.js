import React from 'react';
import { FormattedMessage } from 'react-intl';

import Torrent from 'stores/torrent';

const {
  ERR_TRACKER_WARNING,
  ERR_TRACKER_ERROR,
  ERR_LOCAL_ERROR,
} = Torrent;

function TrackerWarning({ torrent }) {
  const { errorDescription } = torrent;

  return (
    <FormattedMessage
      id='torrent.error.trackerwarning'
      defaultMessage={`Tracker returned a warning: {errorDescription}`}
      values={{errorDescription}}
    />
  );
}

function TrackerError({ torrent }) {
  const { errorDescription } = torrent;

  return (
    <FormattedMessage
      id='torrent.error.trackererror'
      defaultMessage={`Tracker returned an error: {errorDescription}`}
      values={{errorDescription}}
    />
  );
}

function LocalError({ torrent }) {
  const { errorDescription } = torrent;

  return (
    <FormattedMessage
      id='torrent.error.localerror'
      defaultMessage={`Error: {errorDescription}`}
      values={{errorDescription}}
    />
  );
}

export default function({ torrent }) {
  switch (torrent.error) {
    case ERR_TRACKER_WARNING:
      return <TrackerWarning torrent={torrent} />
    case ERR_TRACKER_ERROR:
      return <TrackerError torrent={torrent} />
    case ERR_LOCAL_ERROR:
      return <LocalError torrent={torrent} />
    default:
      return null;
  }
}
