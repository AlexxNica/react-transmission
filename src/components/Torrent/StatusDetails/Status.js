import React from 'react';
import { FormattedMessage } from 'react-intl';

import Torrent from 'stores/torrent';

const {
  STATUS_STOPPED,
  STATUS_CHECK_WAIT,
  STATUS_CHECK,
  STATUS_DOWNLOAD_WAIT,
  STATUS_DOWNLOAD,
  STATUS_SEED_WAIT,
  STATUS_SEED,
} = Torrent;

export default function({ torrent }) {
  switch (torrent.status) {
    case STATUS_CHECK_WAIT:
      return <FormattedMessage id='torrent.status.checkwait' defaultMessage='Queued for verification' />;
    case STATUS_CHECK:
      return <FormattedMessage id='torrent.status.check' defaultMessage='Verifying local data' />;
    case STATUS_DOWNLOAD_WAIT:
      return <FormattedMessage id='torrent.status.downloadwait' defaultMessage='Queued for download' />;
    case STATUS_DOWNLOAD:
      return <FormattedMessage id='torrent.status.download' defaultMessage='Downloading' />;
    case STATUS_SEED_WAIT:
      return <FormattedMessage id='torrent.status.seedwait' defaultMessage='Queued for seeding' />;
    case STATUS_SEED:
      return <FormattedMessage id='torrent.status.seed' defaultMessage='Seeding' />;
    case STATUS_STOPPED:
      return (
      torrent.isFinished
        ? <FormattedMessage id='torrent.status.finished' defaultMessage='Finished' />
        : <FormattedMessage id='torrent.status.paused' defaultMessage='Paused' />
      );
    case null:
    case undefined:
      return <FormattedMessage id='torrent.status.unknown' defaultMessage='Unknown' />;
    default:
      return <FormattedMessage id='torrent.status.error' defaultMessage='Error' />;
  }
}
