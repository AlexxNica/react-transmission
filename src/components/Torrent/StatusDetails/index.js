import React from 'react';

import Seeding from './Seeding';
import Checking from './Checking';
import Downloading from './Downloading';
import Error from './Error';
import Status from './Status';

function StatusDetails({ torrent }) {
  if (torrent.hasErrors) {
    return <Error torrent={torrent} />;
  }

  if (torrent.isDownloading) {
    return <Downloading torrent={torrent} />;
  }

  if (torrent.isSeeding) {
    return <Seeding torrent={torrent} />;
  }

  if (torrent.isChecking) {
    return <Checking torrent={torrent} />;
  }

  return <Status torrent={torrent} />;
}

export default StatusDetails;
