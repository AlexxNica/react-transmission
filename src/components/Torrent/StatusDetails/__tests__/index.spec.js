import React from 'react';
import createComponentWithIntl from 'test/util/createComponentWithIntl';

import StatusDetails from '../';

const torrent = {
  recheckProgress: 1,
  peersGettingFromUs: 5,
  peersSendingToUs: 5,
  peersConnected: 5,
  webseedsSendingToUs: 5,
  rateUpload: 1E5,
  rateDownload: 1E7,
};

test('StatusDetails has errors', () => {
  const component = createComponentWithIntl(
    <StatusDetails torrent={{hasErrors: true}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('StatusDetails downloading', () => {
  const component = createComponentWithIntl(
    <StatusDetails torrent={{...torrent, isDownloading: true}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('StatusDetails seeding', () => {
  const component = createComponentWithIntl(
    <StatusDetails torrent={{...torrent, isSeeding: true}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('StatusDetails checking', () => {
  const component = createComponentWithIntl(
    <StatusDetails torrent={{...torrent, isChecking: true}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('StatusDetails status', () => {
  const component = createComponentWithIntl(
    <StatusDetails torrent={{...torrent}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});
