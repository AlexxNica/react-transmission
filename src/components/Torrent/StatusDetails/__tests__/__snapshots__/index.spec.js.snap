exports[`test StatusDetails checking 1`] = `
<span>
  Verifying local data (100% tested)
</span>
`;

exports[`test StatusDetails downloading 1`] = `
<span>
  Downloading from 5 of 5 peers and 5 web seeds - ↓ 10.00 MB/s ↑ 100 kB/s
</span>
`;

exports[`test StatusDetails has errors 1`] = `null`;

exports[`test StatusDetails seeding 1`] = `
<span>
  Seeding to 5 of 5 connected peers - ↑ 100 kB/s
</span>
`;

exports[`test StatusDetails status 1`] = `
<span>
  Unknown
</span>
`;
